crates/sui-graphql-rpc/tests/dot_move_e2e.rs

```diff
@@ -0,0 +1,441 @@
+                    version,
+                    ..
+                } = object
+                {
+                    if object_type.module.as_str() == "package"
+                        && object_type.name.as_str() == "UpgradeCap"
+                    {
+                        Some(UpgradeCap(*object_id, *version, *digest))
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                }
+            })
+            .unwrap();
+
+        (pkg_id, upgrade_cap)
+    }
+
+    async fn publish_dot_move_package(
+        cluster: &sui_graphql_rpc::test_infra::cluster::NetworkCluster,
+    ) -> (ObjectID, (ObjectID, SequenceNumber)) {
+        let package_path = PathBuf::from(DOT_MOVE_PKG);
+        let tx = cluster
+            .validator_fullnode_handle
+            .test_transaction_builder()
+            .await
+            .publish(package_path)
+            .build();
+
+        let sig = cluster
+            .validator_fullnode_handle
+            .wallet
+            .sign_transaction(&tx);
+
+        let executed = cluster
+            .validator_fullnode_handle
+            .execute_transaction(sig)
+            .await;
+
+        let (mut pkg_id, mut obj_id) = (None, None);
+
+        for object in executed.object_changes.unwrap() {
+            match object {
+                ObjectChange::Published { package_id, .. } => {
+                    pkg_id = Some(package_id);
+                }
+                ObjectChange::Created {
+                    object_id,
+                    object_type,
+                    owner,
+                    ..
+                } => {
+                    if object_type.module.as_str() == "dotmove"
+                        && object_type.name.as_str() == "AppRegistry"
+                    {
+                        let initial_shared_version = match owner {
+                            Owner::Shared {
+                                initial_shared_version,
+                            } => initial_shared_version,
+                            _ => panic!("AppRegistry should be shared"),
+                        };
+
+                        if !owner.is_shared() {
+                            panic!("AppRegistry should be shared");
+                        };
+
+                        obj_id = Some((object_id, initial_shared_version));
+                    }
+                }
+                _ => {}
+            }
+        }
+
+        (pkg_id.unwrap(), obj_id.unwrap())
+    }
+
+    fn name_query(name: &str) -> String {
+        format!(r#"packageByName(name: "{}") {{ address, version }}"#, name)
+    }
+
+    async fn execute_tx(
+        cluster: &sui_graphql_rpc::test_infra::cluster::NetworkCluster,
+        tx: TransactionData,
+    ) -> SuiTransactionBlockResponse {
+        let sig = cluster
+            .validator_fullnode_handle
+            .wallet
+            .sign_transaction(&tx);
```

Comment:
I vaguely remember there being a `sign_and_execute_transaction` to do both of these steps in one, but maybe I'm thinking of some other  API.

Reply:
Yes there was. thanks!